### ISSUE TITLE
C90 Compliance

### ISSUE THREAD

**Issue by kilianmh** [jcorporation/sds#38]
There was an issue in the original library for `C90` 
compliance here: https://github.com/antirez/sds/issues/74

For a utility library like `sds` it would be nice to be 
compliant even with older C standards and it looks like 
there are not that many changes needed. 

What do you think? @jcorporation @dagostinelli
If you like the idea, I can submit a PR.

**Comment by jcorporation** [comment]
I am not very interested for C90 compatibility, but I will merge a PR if it it has no disadvantages.

> sds.c:640:52: warning: ISO C90 does not support ‘long long’ [-Wlong-long]

How would you workaround this?

**Comment by kilianmh** [comment]
> How would you workaround this?

Most issues can be solved with `conditional compilation` such as:

```c
#if defined(__STDC_VERSION__) && __STDC_VERSION__ >= 199901L
    long long num;
#else
    long num;
#endif
```

This will have no runtime disadvantage, but the code will be bloated somewhat.

The only issue where it's not clear what to 
do is that C90 does not support flexible array members. 
We could use a pointer to the array:
```c
struct __attribute__ ((__packed__)) sdshdr5 {
    unsigned char flags; /* 3 lsb of type, and 5 msb of string length */
    #if defined(__STDC_VERSION__) && __STDC_VERSION__ >= 199901L
        char buf[];
    #else
        char *buf;
    #endif

};
```
But then we have to manually allocate and deallocate the memory.

**Comment by jcorporation** [comment]
Sorry for the very late response.

> But then we have to manually allocate and deallocate the memory.

I do not like this.

**Comment by dagostinelli** [comment]
I think that's part of the whole point of being C90 compliant. It doesn't have all the cool stuff from later C versions but it enjoys much wider adoption than later C versions. 

I would think that enjoying wide adoption should be a goal of a C library like this. 

Sadly, C90 was non-negotiable for me. So I moved my project away from sds way back then. 

The fact that I'm writing to tell you that I did is probably rare. Most folks would probably just quietly move along saying "oh well, maybe he will change his mind one day".
